pyrelational/strategies/regression/bald_strategy.py

```diff
@@ -27,18 +27,18 @@ class SoftBALDStrategy(BALDStrategy):
 
     def __init__(
         self,
-        data_manager: GenericDataManager,
-        model: GenericModel,
         temperature: float = 0.5,
     ):
-        super(SoftBALDStrategy, self).__init__(data_manager, model)
+        super(SoftBALDStrategy, self).__init__()
         assert temperature > 0, "temperature parameter should be greater than 0"
         self.T = torch.tensor(temperature)
 
-    def active_learning_step(self, num_annotate: int) -> List[int]:
-        self.model.train(self.l_loader, self.valid_loader)
-        output = self.model(self.u_loader)
+    def active_learning_step(
+        self, num_annotate: int, data_manager: GenericDataManager, model: GenericModel
+    ) -> List[int]:
+        model.train(data_manager.get_labelled_loader(), data_manager.get_validation_loader())
```

Comment:
maybe, we should brainstorm a way to abstract this away.

Reply:
I think we will need to chat about this at more depth as not all methods require the data_manager or model as part of the ative_learning_step.

Reply:
In our last meeting we discussed to pack the training of the model and obtaining uncertainties into a method that gets called for strategies that use this.

Reply:
Updated in b81c244, however I feel that this is starting to abstract away simple one/two liners which may be confusing on its own right.